packages/dd-trace/src/appsec/iast/analyzers/hsts-header-missing-analyzer.js

```diff
@@ -0,0 +1,22 @@
+'use strict'
+
+const { HSTS_HEADER_MISSING } = require('../vulnerabilities')
+const { MissingHeaderAnalyzer } = require('./missing-header-analyzer')
+
+const HSTS_HEADER_NAME = 'Strict-Transport-Security'
+
+class HstsHeaderMissingAnalyzer extends MissingHeaderAnalyzer {
+  constructor () {
+    super(HSTS_HEADER_MISSING, HSTS_HEADER_NAME)
+  }
+  _validateRequestAndResponse (req, res) {
+    const headerToCheck = res.getHeader(HSTS_HEADER_NAME)
```

Comment:
@uurien you did not include the validation for MaxAge you need to obtain max-age (if not then it the same case as not having header) and then compare it not to be equal to -1 or 0

`int i = ivalue.indexOf(MAX_AGE);
			if (i != -1) {
				d.existHeader = true;
				int j = ivalue.indexOf('=', i);
				if (j != -1) {
					int k = ivalue.indexOf(';', j);
					if (k != -1) {
						d.cause = ivalue.substring(j + 1, k);
					}
					else {
						d.cause = ivalue.substring(j + 1);
					}
				}
				else {
					d.cause = "-1";
				}
			}`
			
			`	public boolean isOk() {
		return existHeader && !"-1".equals(cause) && !"0".equals(cause);
	}`